src/main/java/nl/ordina/jobcrawler/service/VacancyService.java

```diff
@@ -23,65 +27,82 @@ public VacancyService(VacancyRepository vacancyRepository, SkillService skillSer
+    public List<Vacancy> findAll() {
         return vacancyRepository.findAll();
     }
 
-    public Optional<Vacancy> getVacancyByID(UUID id) {
-        return vacancyRepository.findById(id);
-    }
 
-    public Set<Vacancy> getVacanciesBySkill(String skill) {
-        return skillService.getVacanciesBySkill(skill);
+    /**
+     * Updates the specified vacancy, identified by its id.
+     *
+     * @param vacancy The vacancy to update.
+     * @return True if the update succeeded, otherwise false.
+     */
+    @Override
+    public boolean update(Vacancy vacancy) {
+        return false;   //Todo: implementing this function
     }
 
-    public List<Vacancy> getVacanciesByBroker(String broker) {
-        return vacancyRepository.findByBrokerEquals(broker);
-    }
 
-    public Optional<Vacancy> getExistingVacancy(String url) {
-        return vacancyRepository.findByVacancyURLEquals(url);
-    }
+    /**
+     * Saves the specified vacancy to the database.
+     *
+     * @param vacancy The vacancy to save to the database.
+     * @return The saved vacancy.
+     * @throws VacancyURLMalformedException if the URL could not be reached.
+     */
+    @Override
+    public Vacancy save(Vacancy vacancy) {
 
+        if (vacancy.hasValidURL()) {    //checking the url, if it is malformed it will throw a VacancyURLMalformedException
+            return vacancyRepository.save(vacancy);
+        } else {
+            throw new VacancyURLMalformedException(vacancy.getVacancyURL());
+        }
 
-    //******** Deleting ********//
-    // Deletes vacancy and accompanying skills and relations from the database //
-    public void delete(UUID id) {
-        Vacancy vacancyToDelete = vacancyRepository.findById(id).orElseThrow(() -> new VacancyNotFoundException(id));
-        Set<Skill> skillsToDelete = new HashSet<Skill>(vacancyToDelete.getSkills()); // saves the skills of the vacancy
+    }
 
-        skillsToDelete.forEach(skill -> skill.removeVacancy(vacancyToDelete)); // remove all vacancy relations
 
-        vacancyRepository.deleteById(id); //deletes the vacancy and all relations to the skills
+    /**
+     * Deletes the vacancy with the specified id.
+     *
+     * @param id The id of the vacancy to delete.
+     * @return True if the operation was successful, false otherwise.
+     * @throws VacancyNotFoundException if the vacancy with the specified id was not found.
+     */
+    @Override
+    public boolean delete(UUID id) {
+//        Vacancy vacancyToDelete = vacancyRepository.findById(id).orElseThrow(() -> new VacancyNotFoundException(id));
+//        Set<Skill> skillsToDelete = new HashSet<Skill>(vacancyToDelete.getSkills()); // saves the skills of the vacancy
+//
+//        skillsToDelete.forEach(skill -> skill.removeVacancy(vacancyToDelete)); // remove all vacancy relations
+
+        try {
+            vacancyRepository.deleteById(id); //deletes the vacancy and all relations to the skills
+        } catch (Exception e) {
```

Comment:
Te globale exception. Hiermee slikt de applicatie alle exceptions in, waardoor exceptions die je niet zou verwachten toch ongemerkt worden "afgehandeld". Vang altijd zo specifiek mogelijk exceptions af. Liefst met loggen van een exception op level WARN of ERROR afhankelijk van hoe ernstig je dat zou vinden. Op die manier is er in ieder geval nog in de log terug te vinden wat er fout ging en kun je achteraf daar nog iets mee doen.